parser/parser_manager.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package parser
+
+import (
+	"archive/tar"
+	"errors"
+	"io"
+)
+
+type ArtifactParser interface {
+	ParseHeader(tr *tar.Reader, hPath string) error
```

Comment:
not sure if this is useful but I can see here reader/writer interfaces that would be composed into parser:

```
type ArticaftParseReader interface {
ParseHeader(tr *tar.Reader, hPath string) error
ParseData(r io.Reader) error
}

type ArticaftParseWriter interface {
ArchiveHeader(tw *tar.Writer, srcDir, dstDir string) error
ArchiveData(tw *tar.Writer, srcDir, dst string) error
}

type ArtifactParser interface {
ArticaftParseWriter
ArticaftParseReader
NeedsDataFile() bool
}
```